core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -400,7 +477,7 @@ public boolean isEmpty() {
 
    @Override
    public boolean containsKey(Object key) {
-      return cacheImplementation.containsKey(key, flags);
+      return get(key) != null;
```

Comment:
Sorry I wrote this too quickly. I can delegate to `CacheImpl.containsKey`, I would need to add some additional methods for `containsKey` and `get` in `CacheImpl`, just more cod than I would have hoped, but seems fine.